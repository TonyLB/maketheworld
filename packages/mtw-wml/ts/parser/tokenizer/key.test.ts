import SourceStream from './sourceStream'
import keyValueTokenizer from './key'

describe('keyValueTokenizer', () => {
    beforeEach(() => {
        jest.clearAllMocks()
        jest.resetAllMocks()
    })

    it('should return undefined when no match', () => {
        const testStream = new SourceStream('Testing')
        expect(keyValueTokenizer(testStream)).toBe(undefined)
    })
    it('should tokenize with whitespace around value', () => {
        const testStream = new SourceStream('( Test )></Room>')
        expect(keyValueTokenizer(testStream)).toMatchSnapshot()
    })
    it('should fail with whitespace within value', () => {
        const testStream = new SourceStream('(Te st)></Room>')
        expect(keyValueTokenizer(testStream)).toMatchSnapshot()
    })
    it('should tokenize without whitespace', () => {
        const testStream = new SourceStream('(Test_1)></Room>')
        expect(keyValueTokenizer(testStream)).toMatchSnapshot()
    })
    it('should reject illegal characters', () => {
        const testStream = new SourceStream('(Test-1)></Room>')
        expect(keyValueTokenizer(testStream)).toMatchSnapshot()
    })
    it('should reject keys that start with a digit', () => {
        const testStream = new SourceStream('(1_Test)></Room>')
        expect(keyValueTokenizer(testStream)).toMatchSnapshot()
    })
})