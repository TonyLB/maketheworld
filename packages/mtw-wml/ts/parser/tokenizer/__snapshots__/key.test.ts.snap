// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keyValueTokenizer should fail with whitespace within value 1`] = `
Object {
  "endIdx": 4,
  "message": "Unexpected token",
  "source": "s",
  "startIdx": 4,
  "type": "Error",
}
`;

exports[`keyValueTokenizer should reject illegal characters 1`] = `
Object {
  "endIdx": 5,
  "message": "Unexpected token",
  "source": "-",
  "startIdx": 5,
  "type": "Error",
}
`;

exports[`keyValueTokenizer should tokenize with whitespace around value 1`] = `
Object {
  "endIdx": 7,
  "source": "( Test )",
  "startIdx": 1,
  "type": "KeyValue",
}
`;

exports[`keyValueTokenizer should tokenize without whitespace 1`] = `
Object {
  "endIdx": 7,
  "source": "(Test_1)",
  "startIdx": 1,
  "type": "KeyValue",
}
`;
